composer.json

```diff
@@ -5,7 +5,7 @@
   "homepage": "http://drupal.org/project/graphql",
   "license": "GPL-2.0+",
   "require": {
-    "webonyx/graphql-php": "^0.12.5"
+    "webonyx/graphql-php": "^14.11.8"
```

Comment:
We're making some major jumps in webonyx/graphql-php version. There are also quite large breaking changes in the library itself.

Even if we can sufficiently adapt the 3.x code then upgrading the library will break anyone's code that extends the GraphQL module and uses helper code from the underlying library.

I think if we want to make drupal/graphql compatible with drupal/core 10.x then we have to do so without upgrading webonyx/graphql-php.

Reply:
It is necessary to upgrade to one of the latest versions of the library. The version you are using is no longer compatible with php version 8.1. This version is required by drupal 10.

This is one of the bugs generated by versions "v.0.12.6" and "v.0.13.9" of the graphql-php library. For this reason it is necessary to upgrade to one of the latest versions of the library.


```Deprecated function: Return type of GraphQL\Type\Definition\Type::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in include() (line 28 of /var/www/html/vendor/webonyx/graphql-php/src/Type/Definition/Type.php).
include('/var/www/html/vendor/webonyx/graphql-php/src/Type/Definition/Type.php') (Line: 480)
Composer\Autoload\includeFile('/var/www/html/vendor/composer/../webonyx/graphql-php/src/Type/Definition/Type.php') (Line: 346)
Composer\Autoload\ClassLoader->loadClass('GraphQL\Type\Definition\Type') (Line: 57)
include('/var/www/html/vendor/webonyx/graphql-php/src/Type/Definition/ObjectType.php') (Line: 480)
Composer\Autoload\includeFile('/var/www/html/vendor/composer/../webonyx/graphql-php/src/Type/Definition/ObjectType.php') (Line: 346)
Composer\Autoload\ClassLoader->loadClass('GraphQL\Type\Definition\ObjectType') (Line: 219)
Drupal\graphql\Plugin\GraphQL\Schemas\SchemaPluginBase->getSchema() (Line: 293)
Drupal\graphql\Plugin\GraphQL\Schemas\SchemaPluginBase->getServer() (Line: 102)
Drupal\graphql\GraphQL\Execution\QueryProcessor->processQuery('default:default', Object) (Line: 38)
Drupal\graphql\GraphQL\Utility\Introspection->introspect('default:default') (Line: 81)
Drupal\graphql\Controller\ExplorerController->viewExplorer('default:default', Object)
call_user_func_array(Array, Array) (Line: 123)```

Reply:
ok, then I think we can go with the library upgrade. Best mention it in the release notes, so that people are aware of potential issues.